test/addon_selection_dialog_test.rb

```diff
@@ -70,6 +64,52 @@
       expect(wrapped_addon.selected?).to eq true
     end
 
+    it "preselects the recommended addons" do
+      addon = addon_generator("recommended" => true)
+
+      expect(Yast::UI).to receive(:UserInput).and_return(:next)
+      registration = double(activated_products: [], get_addon_list: [addon])
+      expect(Registration::Addon).to receive(:selected).and_return([]).at_least(:once)
+
+      expect(Yast::Wizard).to receive(:SetContents) do |_title, content, _help, _back, _next|
+        # check the displayed content
+        term = content.nested_find do |t|
+          t.respond_to?(:value) && t.value == :RichText \
+            && t.params.first.params.first == :items
+        end
+
+        # the check box is checked
+        expect(term.params[1]).to include("checkbox-on.png")
+        expect(term.params[1]).to_not include("checkbox-off.png")
+      end
+
+      subject.run(registration)
+    end
+
+    it "does not preselect the recommended addons if something is already selected" do
+      addon = addon_generator("recommended" => true)
+      # just to have an unknown, but "selected" addon for the Addon.selected call
+      fake_addon = addon_generator
+
+      expect(Yast::UI).to receive(:UserInput).and_return(:next)
+      registration = double(activated_products: [], get_addon_list: [addon])
+      expect(Registration::Addon).to receive(:selected).and_return([fake_addon]).at_least(:once)
+
+      # check the displayed content
+      expect(Yast::Wizard).to receive(:SetContents) do |_title, content, _help, _back, _next|
```

Comment:
np: what about extracting this logic to a method? It is duplicated and it is not that rivial. Anyway, I do not have a strong opinion about it.